pkg/highlight/ftdetect.go

```diff
@@ -1,17 +1,33 @@
 package highlight
 
-import "regexp"
-
-// MatchFiletype will use the list of syntax definitions provided and the filename and first line of the file
-// to determine the filetype of the file
-// It will return the corresponding syntax definition for the filetype
-func MatchFiletype(ftdetect [2]*regexp.Regexp, filename string, firstLine []byte) bool {
-	if ftdetect[0] != nil && ftdetect[0].MatchString(filename) {
-		return true
+// MatchFileName will use the list of syntax definitions provided and the filename to
+// determine the filetype of the file
+func MatchFileName(header *Header, filename string) bool {
+	if header != nil {
+		// index 0 refers to the filename respective filetype rule
```

Comment:
I'd prefer `FtDetect[0]` and `FtDetect[1]` to be normal separate fields of `Header`, with self-explanatory names, rather than obscure array elements, so that these comments would be unnecessary.
Yeah, you're not the one who introduced them as an array, but you're the one who's separated them functionally from each other (for a noble cause) so that now it seems there's no point in keeping them bundled as an array.